### PR TITLE
Breaking changes related to removal of timeout manager

### PR DESCRIPTION
**Pull request by ramonsmits**
- Fix breaking changes related to removal of timeout manager
- Marked code related to non-durable messaging with TODO's
- Fix change related to AT persistence configuration

### CODE REVIEW

File: src/Transport/obsoletes.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿namespace NServiceBus
+{
+    using System;
+
+    /// <summary>Configures native delayed delivery.</summary>
```
Comment (bording) [inline src/Transport/obsoletes.cs:5]: XML comments should be removed from obsoletes. The file can be marked with `#pragma warning disable 1591` / `#pragma warning restore 1591`

File: src/Transport/Config/DelayedDeliverySettings.cs [hunk 2]
```diff
@@ -46,9 +37,6 @@ public void DisableDelayedDelivery()
         {
             // disable delayed delivery
             this.GetSettings().Set(WellKnownConfigurationKeys.DelayedDelivery.DisableDelayedDelivery, true);
```
Comment (bording) [inline src/Transport/Config/DelayedDeliverySettings.cs:39]: This setting and the test about throwing when it's disable means there is more work to done here. It's not the transports responsibility to throw in this scenario now. Instead, we need to alter the `DeliveryConstraints` in  `AzureStorageQueueInfrastructure` when this is called to inform core and let it handle things.

File: src/AcceptanceTests/Sending/When_delaying_messages_and_delayed_delivery_is_disabled.cs [hunk 3]
```diff
@@ -46,7 +46,6 @@ public Endpoint()
                 EndpointSetup<DefaultServer>(config =>
                 {
                     var delayedDelivery = config.UseTransport<AzureStorageQueueTransport>().DelayedDelivery();
-                    delayedDelivery.DisableTimeoutManager();
```
Comment (bording) [inline src/AcceptanceTests/Sending/When_delaying_messages_and_delayed_delivery_is_disabled.cs:49]: I suspect this entire test can be removed once we fix up the `DisableDelayedDelivery` API.

**Review by bording — CHANGES_REQUESTED** [review]

File: src/AcceptanceTests/Sending/When_delaying_messages_and_delayed_delivery_is_disabled.cs (hunk 3 above)
Comment (ramonsmits) [reply src/AcceptanceTests/Sending/When_delaying_messages_and_delayed_delivery_is_disabled.cs:49]: @bording what do you mean, that there will be a test part of the core AT's to replace this?

**Review by bording — APPROVED** [review]

**Pull request merged**